Obstacle_Avoidance.ino

```diff
@@ -0,0 +1,47 @@
+#include <Smartcar.h>
+
+const unsigned short LEFT_ODOMETER_PIN = 2;
+const unsigned short RIGHT_ODOMETER_PIN = 3;
+
+BrushedMotor leftMotor(8, 10, 9);
+BrushedMotor rightMotor(12, 13, 11);
+DifferentialControl control(leftMotor, rightMotor);
+
+GY50 gyroscope(37);
+DirectionlessOdometer leftOdometer(100);
+DirectionlessOdometer rightOdometer(100);
+
+const int TRIGGER_PIN = 40; //D6
+const int ECHO_PIN = 41; //D7
+const unsigned int MAX_DISTANCE = 100;
+
+SR04 front(TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+DistanceCar car(control, leftOdometer, rightOdometer);
+
+void setup() {
+
+  Serial.begin(9600);
+  // Initialize the odometers (they won't work otherwise)
+  leftOdometer.attach(LEFT_ODOMETER_PIN, []() {
+    leftOdometer.update();
+  });
+  rightOdometer.attach(RIGHT_ODOMETER_PIN, []() {
+    rightOdometer.update();
+  });
+
+  car.enableCruiseControl();
+  car.setSpeed(1.0); 
+}
+
+void loop() {
+
+  Serial.println(front.getDistance());
+  car.update(); // Maintain the speed
+  // Stop after moving 1 meter
+  if (front.getDistance() < 15 && front.getDistance() > 0) {
+    car.setSpeed(0);
+  }
+
+  delay(100);

```

Comment:
Why `delay`? This can throw off your `car.update()`! If you want to decrease the refresh rate you should use `millis` and intervals instead of blocking calls. :)

Reply:
I updated the sketch, please check it out